src/provider/controller/ContextHandler.ts

```diff
@@ -31,8 +31,11 @@ export class ContextHandler {
      */
     public async send(window: AppWindow, context: Context): Promise<void> {
         const payload: ReceiveContextPayload = {context};
-
-        return this._apiHandler.dispatch(window.identity, APIToClientTopic.RECEIVE_CONTEXT, payload);
+        if (await window.isReadyToReceiveContext()) {
+            return this._apiHandler.dispatch(window.identity, APIToClientTopic.RECEIVE_CONTEXT, payload);
```

Comment:
👍 on the TODO

Reply:
Also, unlikely to be as big of a problem in this case, since we don't expect any kind of response here, but will tag it anyway as its better to be safe.